frontend/src/host_orchestrator/orchestrator/createcvdbugreportaction.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrator
+
+import (
+	"log"
+	"path/filepath"
+
+	"github.com/google/android-cuttlefish/frontend/src/host_orchestrator/orchestrator/cvd"
+	apiv1 "github.com/google/android-cuttlefish/frontend/src/liboperator/api/v1"
+	"github.com/google/android-cuttlefish/frontend/src/liboperator/operator"
+)
+
+type CreateCVDBugReportActionOpts struct {
+	Paths            IMPaths
+	OperationManager OperationManager
+	ExecContext      cvd.CVDExecContext
+	UUIDGen          func() string
+}
+
+type CreateCVDBugReportAction struct {
+	paths       IMPaths
+	om          OperationManager
+	execContext cvd.CVDExecContext
+	uuidgen     func() string
+}
+
+func NewCreateCVDBugReportAction(opts CreateCVDBugReportActionOpts) *CreateCVDBugReportAction {
+	return &CreateCVDBugReportAction{
+		paths:       opts.Paths,
+		om:          opts.OperationManager,
+		execContext: opts.ExecContext,
+		uuidgen:     opts.UUIDGen,
+	}
+}
+
+func (a *CreateCVDBugReportAction) Run() (apiv1.Operation, error) {
+	if err := createDir(a.paths.CVDBugReportsDir); err != nil {
+		return apiv1.Operation{}, err
+	}
+	uuid := a.uuidgen()
+	if err := createNewDir(filepath.Join(a.paths.CVDBugReportsDir, uuid)); err != nil {
+		return apiv1.Operation{}, err
+	}
+	op := a.om.New()
+	go a.createBugReport(uuid, op)
+	return op, nil
+}
+
+const BugReportZipFileName = "cvd_bugreport.zip"
+
+func (a *CreateCVDBugReportAction) createBugReport(uuid string, op apiv1.Operation) {
+	result := &OperationResult{}
+	filename := filepath.Join(a.paths.CVDBugReportsDir, uuid, "cvd_bugreport.zip")
+	cmd := cvd.NewCommand(
+		a.execContext,
+		[]string{"host_bugreport", "--output=" + filename},
```

Comment:
FYI: This command will block reading from `stdin` if there are more than one group running.

Reply:
good point, made the group name required.